### ISSUE TITLE
[core] Unicode characters not drawn

### ISSUE THREAD

**Issue by overdev** [raysan5/raylib#1095]
Please, before submitting a new issue verify and check:

 - [x] I tested it on latest raylib version from master branch
 - [x] I checked there is no similar issue already reported
 - [x] My code has no errors or misuse of raylib

### Issue description

Raylib now has good support for text formatting and rendering, including utf8 encoding/decoding. In cases where the ability to enter text is desired, though, the API does not provide functions and using the key functions to achieve this might work but not as good as expected (this becomes a specially dificult task for non english languages).

### Code Example

As we can read in the docs, GLFW provides a [callback function](https://www.glfw.org/docs/latest/input_guide.html#input_char) for this, so I suppose there's not much to worry on the raylib side besides wrapping it:

```C
typedef void (*CharacterCallback)(unsigned int codepoint);
//...
RLAPI void SetCharacterCallback(CharacterCallback callback);
```

**Comment by raysan5** [comment]
@overdev `GetKeyPressed()` function should be enough, did you try it?

**Comment by overdev** [comment]
Oh..  I didn't know that! What a shame...

I've just tryed it. I'm still having some issue with character with diacritics. I'm yet to investigate the real cause, cause I'm using the Python binding. The code looks like this:

```Python3
# UPDATE
# ---------------------------
k: int = rl.get_key_pressed()
if k:
    text += chr(k)     # chr(x) returns a single unicode character from the integer x value.
```

It works fine for all common characters. When i try to type characters like Á, Ç, Ñ, i get the question mark in the rendered text.

**Comment by overdev** [comment]
Oh... I didn't know that! What a shame...

I've just tryed it. I'm still having some issue with character with diacritics. I'm yet to investigate the real cause, because I'm using the Python binding. The code looks like this:

```Python
# UPDATE
# ---------------------------
k: int = rl.get_key_pressed()
if k:
    text += chr(k)     # chr(x) returns a single unicode character from the integer x value.
```
It works fine for all common characters. When i try to type characters like Á, Ç, Ñ, i get the question mark in the rendered text.

In the Python code, whenever I pass unicode strings to raylib functions, they get encoded in the binding function to a char* compatible type/format. I also added a utility function that allows to set the text encoding (latin1, cp1252, etc.) and it need to be called just once to set an internal global variable.

In the screenshot below I printed a small test of the `GetKeyPressed()` function. In the game window, on the top, we see the text rendered with `DrawTextEx()`. In the IDE beind the game window, we see in the output window the text printed as each char is typed (the unicode, followed by its char* counterpart between single quotes.

![_raylib_get_key_pressed_issue](https://user-images.githubusercontent.com/9625440/74173264-9cad1d00-4c10-11ea-9b49-60346f8d7c40.png)

The font used in the  is Tahoma.

**Comment by raysan5** [comment]
@overdev Captured characters are captured as Unicode codepoints, I think they should be coded as UTF8 for `DrawText()`. I need to take a closer look to that...

**Comment by raysan5** [comment]
@overdev did you try the `DrawTextCodepoints()`?

**Comment by overdev** [comment]
@raysan Sorry for the delay.

I tested it. For some reason it renders what it seems to be
a `-` character instead of the corresponding glyph.

My test code is below:
```python
# ...
rl.draw_text_codepoint(
    neo,                    # font
    0xe2,                   # 'â' character
    rl.Vector2(300, 32),    # position
    16.0,                   # size
    color_b                 # Color(180, 180, 180, 255)
)
#...
```
Screenshot:
![_raylib_get_key_pressed_issue_draw_codepoint](https://user-images.githubusercontent.com/9625440/75990599-e74e4c00-5ed3-11ea-8012-1a276933b70e.png)

Tryed with codepoints 0xE9 (`é`), 0xF1 (`ñ`) and 0xE2 (`â`); all resulting the
same.

PS.: In your comment above you typed `DrawTextCodePoints()`, in plural. Is this another function or just a typo?

**Comment by raysan5** [comment]
Hi @overdev, just tested it with latest raylib from master branch:
```c
DrawTextCodepoint(GetFontDefault(), 0xe2, (Vector2){ 10, 10 }, 50, RED);
```

<img width="177" alt="Capture" src="https://user-images.githubusercontent.com/5766837/75996128-b2ef8580-5efd-11ea-922b-7b02669030b3.PNG">

It seems to work ok.

**Comment by overdev** [comment]
Great!

Actually, the font used is Tahoma. I was using a font called Neo Latina before, hence the `neo`, then changed to Tahoma to do the tests and forgot to rename the variable...

DrawTexture() result:
![_raylib_get_key_pressed_issue_draw_codepoint2](https://user-images.githubusercontent.com/9625440/76019196-15e21c00-5f00-11ea-985a-ce985ff939f4.png)
(I cannot say whether this is expected or not.)

_FYI, at the moment I'm dependent on the release binaries, I'm temporarily unable to build from master. Any changes you made now I'll be able to give feedback probably on the next release only._

**Comment by raysan5** [comment]
@overdev On the drawn `font.texture` do not exist the glyphs: `â`, `é`, `ñ`. Only _basic_ gryphs are available.

When you load the font you need to request those glyphs if they are required:
```c
// Use: Font LoadFontEx(const char *fileName, int fontSize, int *fontChars, int charsCount);
Font tahoma = LoadFontEx("tahoma.ttf", 18, NULL, 250);  // Requesting 250 glyphs, instead of the 95 default ones: 32(space) -> 126(~)
```

Also note that you can use `int *fontChars` to provide the list of required Unicode characters on font generation.

**Comment by overdev** [comment]
My God, in the end, I was causing the problem. 😞 Shame on me.

The problem was that I completely forgot that I shouldn´t encode the strings to CP1252 (Western-Europe) before passing it to the functions. I'm so used to do it that I did it automaticaly. Also, I changed the codepoint limit from 125 to 250 in the GetKeyPressed() piece of code. Now, with the full char font texture, I can add text input with special characters.

I apologize for the trouble. This "issue" can be closed now.
